Don't create lists if in plaintext mode

SnowyMouse opened the pull request:
fix #1583

I tested this on both desktop and web and it works. I'm assuming there's nothing more to this?

charlag commented on src/gui/editor/Editor.js:
```diff
@@ -103,17 +104,16 @@ export class Editor implements ImageHandler {
-					})
-					createList(blocks, /^1\.\s$/, true) // create an ordered list if a line is started with '1. '
-					createList(blocks, /^\*\s$/, false) // create an ordered list if a line is started with '1. '
+				if (!logins.getUserController().props.sendPlaintextOnly) {
```
Depending on globals in UI components is very bad, I think we should pass it in (maybe that's what comment meant, I'm not sure).

SnowyMouse commented:
I've added a function to Editor that sets whether or not lists are automatically created.

In this case, the only time globals are used is in MailEditor's view function, which it already reads sendPlaintextOnly anyway.

charlag commented on src/gui/editor/Editor.js:
```diff
@@ -96,24 +98,28 @@ export class Editor implements ImageHandler {
-					createList(blocks, /^1\.\s$/, true) // create an ordered list if a line is started with '1. '
-					createList(blocks, /^\*\s$/, false) // create an ordered list if a line is started with '1. '
+				if (this._createsLists) {
+					if (e.which === 32) {
```
I forgot to mention that we have Keys constant and `isKeyPressed()` helper.

SnowyMouse replied:
I went ahead and took care of those. My latest commit is a64ca9977ac27e403e00f3388206f75eb9f332da now.

charlag reviewed (approved):
Yes, this is much better, it's better place to make a decision now.

SnowyMouse commented:
Hey @charlag. Sorry this one took a while - had lots of stuff to do the past couple of weeks.

Anyway, I used isKeyPressed and replaced the `32` with `Keys.SPACE`. Hope that's better!

I also noticed that the comment for the unordered list was wrong, so I fixed that too.

vitoreiji commented:
squashed, rebased and merged in 1982d9f9f86e468e243ade5ec7633b61567c4543
